File: solutions/haskell/pangram/1/src/Pangram.hs
module Pangram (isPangram) where
import Data.Char (toLower) 


isPangram :: String -> Bool
isPangram text = 
  if length ([letter | letter <- ['a'..'z'], not (elem letter (map toLower text))]) == 0 
  then 
    True 
  else 
    False

File: solutions/c/pangram/1/pangram.c
#include "pangram.h"
#include <stdlib.h>


bool is_pangram(const char *sentence) {
  char c;
  int count = 0, i = 0;

  if (sentence == NULL || sentence[0] == '\0')
      return false;

  for (c = 'a'; c <= 'z'; c++) {
    while (sentence[i] != '\0') {
        if (sentence[i] == c || sentence[i] == (c - 32)) {
            count++;
            i = 0;
            break;
        }
        i++;
    }
  }
  
  if (count == 26)
    return true;
  return false;
}
